app/lib/search/index_simple.dart

```diff
@@ -141,6 +146,43 @@ class SimplePackageIndex implements PackageIndex {
     _isReady = true;
     _lastUpdated = new DateTime.now().toUtc();
   }
+
+  void _clearScores() {
+    _healthScores = null;
+    _popularityScores = null;
+  }
+
+  void _initScoresIfNeeded() {
+    if (_healthScores != null) return;
+    _healthScores = <String, double>{};
+    _popularityScores = <String, double>{};
+
+    final healthScorer =
+        new scoring.Summary(_documents.values.map((pd) => pd.health ?? 0.0));
+    final popularityScorer = new scoring.Summary(
+        _documents.values.map((pd) => pd.popularity ?? 0.0));
+    for (PackageDocument doc in _documents.values) {
+      if (doc.health != null) {
+        _healthScores[doc.url] = healthScorer.bezierScore(doc.health) * 100.0;
```

Comment:
@kevmoo has started the score summary to do similar thing to npmjs (at least at that time npmjs was doing it, I'm not able to point to it right now). Kevin, could you please expand on it?